### PR TITLE
Fix ClassTooLargeException when running HTTP package tests 

### PR DESCRIPTION
**Pull request by warunalakshitha**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/32400

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 1]
```diff
@@ -998,23 +988,31 @@ private boolean isBBCoveredInErrorEntry(BIRErrorEntry errorEntry, List<BIRBasicB
...
+     * @return List of operands for source file name, start and end positions
      */
-    private String generatePackageId(PackageID pkg) {
-        return String.format("%s/%s:%s", pkg.orgName.value, pkg.name.value, getMajorVersion(pkg.version.value));
+    private List<BIROperand> generatePositionArgs(BIRPackage pkg, Location location) {
+        String pkgId = generatePackageId(pkg.packageID);
+        BIROperand pkgOperand = generateGlobalConstantOperand(pkg, symbolTable.stringType, pkgId);
+        BIROperand fileNameOperand = generateGlobalConstantOperand(pkg, symbolTable.stringType,
+                                                                   location.lineRange().filePath());
+        BIROperand startPositionOperand = generateGlobalConstantOperand(pkg, symbolTable.intType,
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:1012]: I think we can use local variables for positions? @nadundesilva WDYT?

**Comment by chamil321** [comment]
Tested locally the exception is not occurring.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 2]
```diff
@@ -998,23 +997,52 @@ private boolean isBBCoveredInErrorEntry(BIRErrorEntry errorEntry, List<BIRBasicB
...
+                VarKind.TEMP);
+        BIROperand fileNameOperand = new BIROperand(positionFileVar);
+        BIROperand startPositionOperand = new BIROperand(positionStartVar);
+        BIROperand endPositionOperand = new BIROperand(positionEndVar);
+        ConstantLoad fileNameIns = new ConstantLoad(location, location.lineRange().filePath(),
+                fileNameOperand.variableDcl.type, fileNameOperand);
+        ConstantLoad startPositionIns = new ConstantLoad(location, location.lineRange().startLine().line() + 1,
+                startPositionOperand.variableDcl.type, startPositionOperand);
+        ConstantLoad endPositionIns = new ConstantLoad(location, location.lineRange().startLine().offset() + 1,
+                endPositionOperand.variableDcl.type, endPositionOperand);
```
Comment (nadundesilva) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:1039]: IINM, this is the `startLine` and `startCol` isn't it ?
- location.lineRange().startLine().line() - start line
- location.lineRange().startLine().offset() - start column

So shall we update it ? (I think it's there in other places as well)

Comment (warunalakshitha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java]: Fixed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java (hunk 1 above)
Comment (warunalakshitha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java]: I have changed to use local vars.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 3]
```diff
@@ -998,23 +997,52 @@ private boolean isBBCoveredInErrorEntry(BIRErrorEntry errorEntry, List<BIRBasicB
...
      */
-    private String generatePackageId(PackageID pkg) {
-        return String.format("%s/%s:%s", pkg.orgName.value, pkg.name.value, getMajorVersion(pkg.version.value));
+    private List<BIROperand> generatePositionArgs(BIRPackage pkg, BIRFunction func, BIRBasicBlock currentBB,
+                                                  Location location) {
+        String pkgId = generatePackageId(pkg.packageID);
+        BIROperand pkgOperand = generateGlobalConstantOperand(pkg, symbolTable.stringType, pkgId);
+        Name fileName = new Name("$observabilityFileName" + localVarIndex++);
+        Name startLine = new Name("$observabilityStart" + localVarIndex++);
+        Name startCol = new Name("$observabilityEnd" + localVarIndex++);
```
Comment (nadundesilva) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:1024]:
```suggestion
        Name startLine = new Name("$observabilityStartLine" + localVarIndex++);
        Name startCol = new Name("$observabilityStartCol" + localVarIndex++);
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java [hunk 4]
```diff
@@ -998,23 +997,52 @@ private boolean isBBCoveredInErrorEntry(BIRErrorEntry errorEntry, List<BIRBasicB
...
+        BIRVariableDcl positionFileVar = new BIRVariableDcl(symbolTable.stringType, fileName, VarScope.FUNCTION,
+                VarKind.TEMP);
+        BIRVariableDcl positionStartVar = new BIRVariableDcl(symbolTable.intType, startLine, VarScope.FUNCTION,
+                VarKind.TEMP);
+        BIRVariableDcl positionEndVar = new BIRVariableDcl(symbolTable.intType, startCol, VarScope.FUNCTION,
+                VarKind.TEMP);
```
Comment (nadundesilva) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmObservabilityGen.java:1030]:
```suggestion
        BIRVariableDcl positionStartLineVar = new BIRVariableDcl(symbolTable.intType, startLine, VarScope.FUNCTION,
                VarKind.TEMP);
        BIRVariableDcl positionStartColVar = new BIRVariableDcl(symbolTable.intType, startCol, VarScope.FUNCTION,
                VarKind.TEMP);
```

**Review by nadundesilva — APPROVED** [review]
LGTM

**Pull request merged**